Flaky LineTcpReceiverTest.testRenameTable() test

### Describe the bug

Stack trace : 

2023-02-14T13:43:33.8608565Z 2023-02-14T13:43:33.848674Z I i.q.c.TableReader open partition /tmp/junit12022033930939766998/dbRoot/weather~2/2016-06-13 [rowCount=2, partitionNameTxn=-1, transientRowCount=2, partitionIndex=0, partitionCount=1]
2023-02-14T13:43:33.8609559Z 2023-02-14T13:43:33.848823Z E i.q.c.l.t.AbstractLineTcpReceiverTest Stopping ILP worker pool because of an error
2023-02-14T13:43:33.8610197Z java.lang.AssertionError: expected:<location	source	temp	timestamp
2023-02-14T13:43:33.8610797Z east5	sensor2	50.0	2016-06-13T17:43:50.100800Z
2023-02-14T13:43:33.8611375Z west6	sensor3	60.0	2016-06-13T17:43:50.100900Z
2023-02-14T13:43:33.8611934Z north	sensor4	70.0	2016-06-13T17:43:50.101000Z
2023-02-14T13:43:33.8612409Z > but was:<location	source	temp	timestamp
2023-02-14T13:43:33.8613398Z east5	sensor2	50.0	2016-06-13T17:43:50.100800Z
2023-02-14T13:43:33.8613987Z west6	sensor3	60.0	2016-06-13T17:43:50.100900Z
2023-02-14T13:43:33.8614398Z >
2023-02-14T13:43:33.8614800Z 	at org.junit.Assert.fail(Assert.java:89)
2023-02-14T13:43:33.8615305Z 	at org.junit.Assert.failNotEquals(Assert.java:835)
2023-02-14T13:43:33.8615871Z 	at org.junit.Assert.assertEquals(Assert.java:120)
2023-02-14T13:43:33.8616463Z 	at io.questdb.test.tools.TestUtils.assertEquals(TestUtils.java:382)
2023-02-14T13:43:33.8617070Z 	at io.questdb.test.tools.TestUtils.assertEquals(TestUtils.java:365)
2023-02-14T13:43:33.8617675Z 	at io.questdb.test.tools.TestUtils.assertCursor(TestUtils.java:121)
2023-02-14T13:43:33.8618295Z 	at io.questdb.cairo.AbstractCairoTest.assertCursor(AbstractCairoTest.java:524)
2023-02-14T13:43:33.8618954Z 	at io.questdb.cairo.AbstractCairoTest.assertCursorTwoPass(AbstractCairoTest.java:528)
2023-02-14T13:43:33.8619676Z 	at io.questdb.cutlass.line.tcp.AbstractLineTcpReceiverTest.assertTable(AbstractLineTcpReceiverTest.java:226)
2023-02-14T13:43:33.8620457Z 	at io.questdb.cutlass.line.tcp.LineTcpReceiverTest.lambda$testRenameTable$26(LineTcpReceiverTest.java:706)
2023-02-14T13:43:33.8621252Z 	at io.questdb.cutlass.line.tcp.AbstractLineTcpReceiverTest.lambda$runInContext$2(AbstractLineTcpReceiverTest.java:272)
2023-02-14T13:43:33.8622022Z 	at io.questdb.cairo.AbstractCairoTest.lambda$assertMemoryLeak$4(AbstractCairoTest.java:276)
2023-02-14T13:43:33.8622688Z 	at io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:541)
2023-02-14T13:43:33.8623327Z 	at io.questdb.cairo.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:273)
2023-02-14T13:43:33.8624046Z 	at io.questdb.cutlass.line.tcp.AbstractLineTcpReceiverTest.runInContext(AbstractLineTcpReceiverTest.java:264)
2023-02-14T13:43:33.8624797Z 	at io.questdb.cutlass.line.tcp.LineTcpReceiverTest.testRenameTable(LineTcpReceiverTest.java:680)
2023-02-14T13:43:33.8625439Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-02-14T13:43:33.8626071Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-02-14T13:43:33.8627173Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-02-14T13:43:33.8627898Z 	at java.lang.reflect.Method.invoke(Method.java:566)
2023-02-14T13:43:33.8628522Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-02-14T13:43:33.8629210Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-02-14T13:43:33.8629882Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-02-14T13:43:33.8630554Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-02-14T13:43:33.8631217Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-02-14T13:43:33.8631866Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2023-02-14T13:43:33.8632582Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2023-02-14T13:43:33.8633169Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2023-02-14T13:43:33.8633827Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
2023-02-14T13:43:33.8634644Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
2023-02-14T13:43:33.8635307Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
2023-02-14T13:43:33.8635832Z 	at java.lang.Thread.run(Thread.java:829)

### To reproduce

_No response_

### Expected Behavior

_No response_

### Environment

```markdown
- **QuestDB version**:
- **OS**:
- **Browser**:
```


### Additional context

_No response_

--- a/core/src/test/java/io/questdb/cutlass/line/tcp/AbstractLineTcpReceiverTest.java
+++ b/core/src/test/java/io/questdb/cutlass/line/tcp/AbstractLineTcpReceiverTest.java
@@ -310,7 +310,7 @@ public class AbstractLineTcpReceiverTest extends AbstractCairoTest {
                 break;
             case WAIT_ILP_TABLE_RELEASE:
                 receiver.setSchedulerListener((tableName1, event) -> {
-                    if (tableName1 != null && tablesToWaitFor.remove(tableName1.getTableName()) != null) {
+                    if (event == PoolListener.EV_RETURN && tableName1 != null && tablesToWaitFor.remove(tableName1.getTableName()) != null) {
                         releaseLatch.countDown();
                     }
                 });

--- a/core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java
+++ b/core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java
@@ -350,7 +350,7 @@ public class LineTcpReceiverTest extends AbstractLineTcpReceiverTest {
                     weather + ",location=us-eastcoast,source=sensor2 temp=89 1465839830102400200\n" +
                     weather + ",location=us-westcost,source=sensor1 temp=82 1465839830102500200\n" +
                     "done ok=t\n";
-            sendLingerWaitILPRelease(receiver, lineData, weather, "done");
+            sendWaitWalReleaseCount(receiver, lineData, 3);
 
             mayDrainWalQueue();
 
@@ -687,7 +687,7 @@ public class LineTcpReceiverTest extends AbstractLineTcpReceiverTest {
                     weather + ",location=north,source=sensor4 temp=70 1465839830101000200\n" +
                     meteorology + ",location=south temperature=80 1465839830101000200\n";
 
-            sendLingerWaitILPRelease(receiver, lineData, weather, meteorology);
+            sendWaitWalReleaseCount(receiver, lineData, 3);
 
             mayDrainWalQueue();
 
@@ -741,7 +741,7 @@ public class LineTcpReceiverTest extends AbstractLineTcpReceiverTest {
                     weather + ",location=south7 temperature=70 1465839830101000200\n" +
                     meteorology + ",location=south8 temperature=80 1465839830101000200\n";
 
-            sendLingerWaitILPRelease(receiver, lineData, weather, meteorology);
+            sendWaitWalReleaseCount(receiver, lineData, 3);
 
             mayDrainWalQueue();
             // two of the three commits go to the renamed table
@@ -1651,6 +1651,19 @@ public class LineTcpReceiverTest extends AbstractLineTcpReceiverTest {
 
     private void sendNoWait(LineTcpReceiver receiver, String lineData, String tableName) {
         send(receiver, lineData, tableName, WAIT_NO_WAIT);
+    }
+
+    private void sendWaitWalReleaseCount(LineTcpReceiver receiver, String lineData, int walReleaseCount) {
+        SOCountDownLatch releaseLatch = new SOCountDownLatch(walReleaseCount);
+        receiver.setSchedulerListener((tableName1, event) -> {
+            if (event == PoolListener.EV_RETURN && tableName1 != null) {
+                LOG.info().$("=== released WAL writer === ").$(tableName1.getDirName()).$(":").$(tableName1.getTableName()).$();
+                releaseLatch.countDown();
+            }
+        });
+
+        send(receiver, lineData, "dummy", WAIT_NO_WAIT);
+        releaseLatch.await(10 * Timestamps.SECOND_MICROS * 1000L);
     }
 
     private void shutdownReceiverWhileSenderIsSendingData(WorkerPool ioPool, WorkerPool writerPool) throws SqlException {

